sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/PartitionMapper.java

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.implementation.Strings;
+import com.azure.cosmos.implementation.apachecommons.lang.tuple.Pair;
+import com.azure.cosmos.implementation.feedranges.FeedRangeEpkImpl;
+import com.azure.cosmos.implementation.routing.Range;
+
+import java.util.ArrayDeque;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public final class PartitionMapper {
+    public static <T extends IPartitionedToken> PartitionMapping<T> getPartitionMapping(
```

Comment:
I don't see any tests for this class. It's a big part of split and merge proofing the continuation token, so I would hope to see more.

Reply:
Ported the ContinuationResumeLogicTests https://github.com/mbhaskar/azure-sdk-for-java/blob/3ae027ae2beb2506b8e74c256224e5156e1e740d/sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/query/ContinuationResumeLogicTests.java#L19